wallet-new/src/Cardano/Wallet/WalletLayer/Legacy.hs

```diff
@@ -1,43 +1,233 @@
+                                <*> pure 0
+
+    let walletInit = CWalletInit initMeta backupPhrase
+
+    wallet      <- newWalletHandler newwalOperation spendingPassword walletInit
+    wId         <- migrate $ cwId wallet
+
+    -- Get wallet or throw if missing.
+    maybeThrow (MissingWallet wId) =<< pwlGetWallet wId
+  where
+    -- | We have two functions which are very similar.
+    newWalletHandler :: WalletOperation -> PassPhrase -> CWalletInit -> m CWallet
+    newWalletHandler CreateWallet  = newWallet
+    newWalletHandler RestoreWallet = restoreWallet
+
+
+pwlGetWalletIds
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => m [WalletId]
+pwlGetWalletIds = do
+    ws          <- askWalletSnapshot
+    migrate $ getWalletAddresses ws
+
+pwlGetWallet
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> m (Maybe Wallet)
+pwlGetWallet wId = do
+    ws          <- askWalletSnapshot
+
+    cWId        <- migrate wId
+    wallet      <- V0.getWallet cWId
+
+    pure $ do
+        walletInfo  <- getWalletInfo cWId ws
+        migrate (wallet, walletInfo)
+
+pwlUpdateWallet
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> WalletUpdate
+    -> m Wallet
+pwlUpdateWallet wId wUpdate = do
+    walletDB    <- askWalletDB
+
+    cWId        <- migrate wId
+    cWMeta      <- migrate wUpdate
+
+    -- Update the data
+    setWalletMeta walletDB cWId cWMeta
+
+    -- Get wallet or throw if missing.
+    maybeThrow (MissingWallet wId) =<< pwlGetWallet wId
+
+-- | Seems silly, but we do need some sort of feedback from
+-- the DB.
+pwlDeleteWallet
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> m Bool
+pwlDeleteWallet wId = do
+    cWId        <- migrate wId
+    -- TODO(ks): It would be better to catch specific @Exception@.
+    -- Maybe @try@?
+    catchAll (const True <$> V0.deleteWallet cWId) (const . pure $ False)
+
+------------------------------------------------------------
+-- Account
+------------------------------------------------------------
+
+pwlCreateAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> NewAccount
+    -> m Account
+pwlCreateAccount wId newAcc@NewAccount{..} = do
+
+    let spendingPassword = fromMaybe mempty . fmap coerce $ naccSpendingPassword
+
+    accInit     <- migrate (wId, newAcc)
+    cAccount    <- V0.newAccount RandomSeed spendingPassword accInit
+
+    migrate cAccount
+
+pwlGetAccounts
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> m [Account]
+pwlGetAccounts wId = do
+    cWId        <- migrate wId
+    cAccounts   <- V0.getAccounts $ Just cWId
+    migrate cAccounts
+
+pwlGetAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> AccountIndex
+    -> m (Maybe Account)
+pwlGetAccount wId aId = do
+    accId       <- migrate (wId, aId)
+    account     <- V0.getAccount accId
+    Just <$> migrate account
```

Comment:
`V0.getAccount` throws whenever the account is not found. Yet, the function signature suggests that we'd return `Nothing` in such case, am I right? 
Perhaps we can recover from the exception thrown by V0 methods `getAccountAddrsOrThrow` (and friends) and leverage the `Maybe`? What do you think :no_mouth: ?